dipy/segment/tests/test_metric.py

```diff
@@ -174,7 +174,7 @@ def dist(self, v1, v2):
             if metric.are_compatible(f1.shape, f2.shape):
                 distance = metric.dist(f1, f2)
                 if np.all(f1 == f2):
-                    assert_equal(distance, 0.)
+                    assert_almost_equal(distance, 0.)
```

Comment:
There is no check verifying if the two vectors are equal in the distance function. My guess is that there is a loss of precision at lines [141](https://github.com/nipy/dipy/pull/686/files#diff-e5143f1d5de219e5851a71755ddae0b4L141)-[143](https://github.com/nipy/dipy/pull/686/files#diff-e5143f1d5de219e5851a71755ddae0b4L143), `cos_theta` must be less than 1.
```python
v1_normed = v1.astype(np.float64) / norm(v1.astype(np.float64))
v2_normed = v2.astype(np.float64) / norm(v2.astype(np.float64))
cos_theta = np.dot(v1_normed, v2_normed.T)
```
Surely, I could add an equality test but I am afraid that adding an `if` in this critical part would slow down the clustering (I didn't do any benchmarks though).